src/tests/ftest/fault_domain/fault_domain.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+from apricot import TestWithServers
+
+
+class FaultDomain(TestWithServers):
+    """Test class Fault domains
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up fault domain test."""
+        self.start_agents_once = False
+        self.start_servers_once = False
+        self.setup_start_agents = False
+        self.setup_start_servers = False
+        super().setUp()
+
+    def test_pools_in_different_domains(self):
+        """The aims to:
+            Be able to configure daos servers using different fault domains.
+            Created pools must be in different fault domains.
+
+           Not coded:
+            If a pool has the same fault domains, a warning should be expected.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=fault_domain,fault_domain_different_domains
+        """
+        test_passed = True
+        error_messages = list()
+        # Test pools is a list of OrderedDicts
+        # Here is an example of the first element of test_pools:
+        # OrderedDict([('pool_0',
+        #               OrderedDict([('name', 'daos_server'),
+        #                            ('size', '9G'),
+        #                            ('nranks', 1),
+        #                            ('control_method', 'dmg')]))])
+        test_pools = self.params.get("test_pools", "/run/*")
+        fault_paths = self.params.get("fault_path", '/run/*')
+        test_servers = self.hostlist_servers
+
+        for counter, server in enumerate(test_servers):
+            self.add_server_manager()
+            self.configure_manager(
+                "daos_server",
+                self.server_managers[counter],
+                [server],
+                self.hostfile_servers_slots
+            )
+            self.server_managers[counter].set_config_value("fault_path",
+                                                           fault_paths[counter])
+
+        # Servers are started
+        for server in self.server_managers:
+            server.start()
+
+        self.start_agents()
+
+        # Create pools, setting the values obtained from yaml file
+        self.pool = []
+        for index in range(len(test_pools)):
```

Comment:
(pylint-consider-using-enumerate) Consider using enumerate instead of iterating with range and len